package gap.msgs;

/// \ingroup gap_msgs
/// \interface CameraUtilsRequest
/// \brief A request for Camera Utils plugin

import "pose.proto";
import "point_projection.proto";

message CameraUtilsRequest
{
    /// \brief Type of request 
    enum Type
    {
        /// \brief Acquire frame and save it to disk 
        CAPTURE     = 1;
        /// \brief Change output directory 
        DIR         = 2;
        /// \brief Move camera
        MOVE        = 3;
        /// \brief Project 3D point to 2D in camera plane
        PROJECTION  = 4;
    }

    /// \brief Type of request 
    optional Type               type        = 1;
    /// \brief File path of captured frame
    optional string             file_name   = 2;
    /// \brief Path to the output folder 
    optional string             output_dir  = 3;
    /// \brief Expected camera pose
    optional gazebo.msgs.Pose   pose        = 4;
    /// \brief 3D to 2D point projection 
    repeated PointProjection    projections = 5;
}
